protocol/app/upgrades_test.go

```diff
@@ -0,0 +1,80 @@
+package app_test
+
+import (
+	"errors"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	"github.com/dydxprotocol/v4-chain/protocol/app"
+	"github.com/dydxprotocol/v4-chain/protocol/app/upgrades"
+	testapp "github.com/dydxprotocol/v4-chain/protocol/testutil/app"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSetupUpgradeHandlers(t *testing.T) {
+	upgradeName := "test-upgrade"
+	createHandlerWasCalled := 0
```

Comment:
IIUC, these vars are shared across different test cases. I think it's better to isolate these vars per test case so the outcome of this test does not depend on the sequence of the test run (which is non-deterministic when iterating maps)